lib/puppet/transaction/persistence.rb

```diff
@@ -56,18 +56,21 @@ def load
       begin
         result = Puppet::Util::Yaml.load_file(filename)
       rescue Puppet::Util::Yaml::YamlLoadError => detail
-        Puppet.err "Transaction store file #{filename} is corrupt (#{detail}); replacing"
+        Puppet.log_exception(detail, "Transaction store file #{filename} is corrupt (#{detail}); replacing")
 
         begin
           File.rename(filename, filename + ".bad")
-        rescue
+        rescue => detail
+          Puppet.log_exception(detail, "Unable to rename corrupt transaction store file: #{detail}")
           raise Puppet::Error, "Could not rename corrupt transaction store file #{filename}; remove manually", detail.backtrace
         end
+
+        result = {}
       end
     end
 
     unless result.is_a?(Hash)
-      Puppet.err "Transaction state file #{filename} is valid YAML but not returning a hash. Check the file for corruption, or remove it before continuing."
+      Puppet.err "Transaction store file #{filename} is valid YAML but not returning a hash. Check the file for corruption, or remove it before continuing."
```

Comment:
I assume this should stay an error, because if we hit this we've gone well past the new line 68 above and something's really gone horribly wrong.